### PR TITLE
Fix numpy error with array edge_width in Markers

### PR DESCRIPTION
**Pull request by brisvag**
When passing an array to `Markers.edge_width`, a `ValueError` is raised:

```python
    if edge_width < 0:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

```

This PR fixes this by using `np.all`.

Also, if a `list` is passed, the comparison fails because

```python
TypeError: '<' not supported between instances of 'list' and 'int'
```

Therefore, I added a line to cast to array first.

### CODE REVIEW

File: vispy/visuals/markers.py [hunk 1]
```diff
@@ -673,11 +673,14 @@ def set_data(self, pos=None, size=10., edge_width=1., edge_width_rel=None,
         if (edge_width is not None) + (edge_width_rel is not None) != 1:
             raise ValueError('exactly one of edge_width and edge_width_rel '
                              'must be non-None')
+
         if edge_width is not None:
-            if edge_width < 0:
+            edge_width = np.asarray(edge_width)
+            if np.all(edge_width < 0):
```
Comment (djhoese) [inline vispy/visuals/markers.py:679]: Should be `np.any` shouldn't it?

**Review by djhoese — CHANGES_REQUESTED** [review]
:confused: How did this ever work?

**Comment by brisvag** [comment]
> :confused: How did this ever work?

We don't use it yet in napari; also, I think it became an error recently and used to be a warning.

**Comment by djhoese** [comment]
Can you add a test?

**Comment by brisvag** [comment]
Is build wheels stuck?

**Comment by djhoese** [comment]
I'll restart it.

**Comment by djhoese** [comment]
Looking at the raw log it looks like the Python 3.8 wheels hang when they are being tested. In the past I've seen wheel builds hang/fail when dependency wheels aren't available (like numpy) and the CI job here tries to build the package from source. But in this case this is Python 3.8 and happening during the testing. I guess I'd assume it is some dependency causing the hang up, but I'm not really sure where to go from here.

**Comment by brisvag** [comment]
Yeah... Odd :/

**Review by almarklein — APPROVED** [review]

**Pull request merged**